### DISCUSSION TITLE
CR-Sparse v0.3.2

### DISCUSSION THREAD

**Discussion by shailesh1729** [carnotresearch/cr-sparse#33] [Announcements]
[Documentation](https://cr-sparse.readthedocs.io/en/v0.3.2/)

### Added

Dictionaries

- Grassmannian frames

Linear operators

- windowed_op

Sparse Recovery algorithms

- FOCUSS
- SPGL1 (Spectral Projected Gradient L1)

Optimization : Smooth functions

- smooth_quad_error 

Optimization algorithms

- Spectral projected gradient

Test Problems

- New test problems module introduced
- heavi-sine:fourier:heavi-side
- blocks:haar
- cosine-spikes:dirac-dct
- complex:sinusoid-spikes:dirac-fourier
- cosine-spikes:dirac-dct:gaussian
- piecewise-cubic-poly:daubechies:gaussian
- signed-spikes:dirac:gaussian
- complex:signed-spikes:dirac:gaussian
- blocks:heavi-side
- blocks:normalized-heavi-side
- gaussian-spikes:dirac:gaussian
- src-sep-1

Examples

- Matching pursuit demo
- Grassmannian frames demo
- Several examples based on the test problems

Documentation

- Thinking in JAX tutorial added
- Quick start expanded
- Test problems documentation linked with examples

### Changed

- Matching pursuit implementation revamped

### Fixed

- Handling of complex signals in Subspace Pursuit
- Handling of complex signals in Compressive Sampling Matching Pursuit

### Improved

- Support change condition added in convergence criteria for Subspace Pursuit
- order attribute in reshape linear operator

### Removed

- `cr.sparse.io` moved to `cr-nimble` project

<hr /><em>This discussion was created from the release <a href='https://github.com/carnotresearch/cr-sparse/releases/tag/v0.3.2'>CR-Sparse v0.3.2</a>.</em>

**Comment by shailesh1729** [comment]
The story behind this release is [here](https://www.indigits.com/post/2022/10/crsparse_032).

**Comment by Phoenix1017** [comment]
您好，已收到亲的邮件。我将尽快查阅。-------------------------------------------------------市场调查合作伙伴！负责整个深圳区域的执行。黄楷杰Tel: 13662628489
